integration/container_volume_test.go

```diff
@@ -0,0 +1,170 @@
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package integration
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	goruntime "runtime"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	runtime "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+)
+
+func createRegularFile(basePath, content string) (string, error) {
+	newFolder := filepath.Join(basePath, "regular")
+	err := os.Mkdir(newFolder, 0755)
+	if err != nil {
+		return "", err
+	}
+
+	newFile := filepath.Join(newFolder, "foo.txt")
+	err = ioutil.WriteFile(newFile, []byte(content), 0644)
+	return filepath.Join("regular", "foo.txt"), err
+}
+
+func fileInSymlinkedFolder(basePath, targetFile string) (string, error) {
+	symlinkFolder := filepath.Join(basePath, "symlink_folder")
+	err := os.Symlink(filepath.Dir(targetFile), symlinkFolder)
+
+	return filepath.Join(symlinkFolder, filepath.Base(targetFile)), err
+}
+
+func symlinkedFile(basePath, targetFile string) (string, error) {
+	symlinkFile := filepath.Join(basePath, "symlink_file")
+	err := os.Symlink(targetFile, symlinkFile)
+
+	return symlinkFile, err
+}
+
+func symlinkedFileInSymlinkedFolder(basePath, targetFile string) (string, error) {
+	symlinkFolder := filepath.Join(basePath, "symlink_folder")
+	err := os.Symlink(filepath.Dir(targetFile), symlinkFolder)
+	if err != nil {
+		return "", err
+	}
+
+	symlinkFile := filepath.Join(basePath, "symlink_file")
+	err = os.Symlink(filepath.Join(symlinkFolder, filepath.Base(targetFile)), symlinkFile)
+
+	return symlinkFile, err
+}
+
+func TestContainerSymlinkVolumes(t *testing.T) {
+	for name, test := range map[string]struct {
+		createFile func(basePath, targetFile string) (string, error)
+	}{
+		// Create difference file / symlink scenarios:
+		// - symlink_file -> regular_folder/regular_file
+		// - symlink_folder/regular_file (symlink_folder -> regular_folder)
+		// - symlink_file -> symlink_folder/regular_file (symlink_folder -> regular_folder)
+		"file in symlinked folder": {
+			createFile: fileInSymlinkedFolder,
+		},
+		"symlinked file": {
+			createFile: symlinkedFile,
+		},
+		"symlinkedFileInSymlinkedFolder": {
+			createFile: symlinkedFileInSymlinkedFolder,
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			testPodLogDir, err := ioutil.TempDir("/tmp", "symlink-test")
+			require.NoError(t, err)
+			defer os.RemoveAll(testPodLogDir)
+
+			testVolDir, err := ioutil.TempDir("/tmp", "symlink-test-vol")
+			require.NoError(t, err)
+			defer os.RemoveAll(testVolDir)
```

Comment:
TempDir uses the OS's default temporary directory if the first parameter is `""`. Do you specifically want to use `/tmp` here?